File: Threads.java
public class Threads implements Runnable {
	static final int THREADS = 2;
	int id;

	public Threads(int id) {
		this.id = id;
	}

	@Override
	public void run() {
		System.out.printf("Hi, I'm thread %d\n", id);
	}

	public static void main(String[] args) throws InterruptedException {
		Thread[] threads = new Thread[THREADS];
		int i;
		
		for (i=0; i< THREADS; i++) {
			threads[i] = new Thread(new Threads(i));
			threads[i].start();
		}

		for (i=0; i< THREADS; i++) {
			threads[i].join();
		}
		
	}
}

File: threads.py
import threading
import time

THREADS = 2

class myThread(threading.Thread):
	def __init__(self, threadID):
		threading.Thread.__init__(self)
		self.threadID = threadID

	def run(self):
		print("Hi, I'm thread {}".format(self.threadID))

threads = []

for i in range(THREADS):
	# Create new threads
	t = myThread(i)
	threads.append(t)
	t.start() # start the thread

# Wait for all threads to complete
for t in threads:
	t.join()


